client/battle/CBattleInterface.cpp

```diff
@@ -1130,48 +1138,45 @@ void CBattleInterface::newRound(int number)
 	console->addText(CGI->generaltexth->allTexts[412]);
 }
 
-void CBattleInterface::giveCommand(Battle::ActionType action, BattleHex tile, ui32 stackID, si32 additional, si32 selected)
+void CBattleInterface::giveCommand(EActionType action, BattleHex tile, si32 additional)
 {
-	const CStack *stack = curInt->cb->battleGetStackByID(stackID);
-	if (!stack && action != Battle::HERO_SPELL && action != Battle::RETREAT && action != Battle::SURRENDER)
+	const CStack * actor = nullptr;
+	if(action != EActionType::HERO_SPELL && action != EActionType::RETREAT && action != EActionType::SURRENDER)
 	{
-		return;
+		actor = activeStack;
 	}

```

Comment:
I think that addidional actor pointer that point to activeStack is not needed, you can move that `if` to top of sendCommand and use activeStack